views/textarea.go

```diff
@@ -91,14 +90,19 @@ func (m *linesModel) GetCursor() (int, int, bool, bool) {
 func (ta *TextArea) SetLines(lines []string) {
 	ta.Init()
 	m := ta.model
-	m.width = 0
-	m.height = len(lines)
-	m.lines = append([]string{}, lines...)
-	for _, l := range lines {
-		if len(l) > m.width {
-			m.width = len(l)
+
+	for row, line := range lines {
+		// extend slice before using m.runes[row] to avoid panic
+		m.runes = append(m.runes, []rune{})
```

Comment:
Do this outside of the loop, or in initialization code elsewhere (not sure off the top of my head if we have an allocator, we probably do.).

Reply:
Moved out of the loop. I have to keep it in this function as number or lines is unknown before.